### PR TITLE
Update most packages

### PR DESCRIPTION
**Pull request by hoodmane**
Except:

* geopandas -- we're running too many tests after the update so it times out, have to change the geopandas tests to do less.
* lightgbm
* matplotlib #4510
* numcodecs
* numpy: the release from this week fails to build #4498
* pillow_heif
* protocol_buffer
* pyinstrument
* python-sat
* rebound
* reboundx
* river
* scipy (#4499)
* shapely (#4503)
* sparseqr
* zengl (#4509)

### CODE REVIEW

File: packages/shapely/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
   name: shapely
-  version: 1.8.2
+  version: 2.0.2
   top-level:
     - shapely
 source:
-  url: https://files.pythonhosted.org/packages/93/3c/cda77e57a08c49569de5bd90376c547bcb981420100adcb0f3770ed681b1/Shapely-1.8.2.tar.gz
-  sha256: 572af9d5006fd5e3213e37ee548912b0341fb26724d6dc8a4e3950c10197ebb6
-  patches:
-    - patches/0001-Search-libs-from-auditwheel-repaired-directory.patch
```
Comment (hoodmane) [inline packages/shapely/meta.yaml:10]: @ryanking13 Maybe dropping this patch is going to be a problem?

File: packages/pydantic/meta.yaml [hunk 2]
```diff
@@ -1,11 +1,11 @@
 package:
   name: pydantic
-  version: 1.10.7
+  version: 2.6.1
   top-level:
     - pydantic
 source:
-  sha256: cfc83c0678b6ba51b0532bea66860617c4cd4251ecf76e9846fa5a9f3454e97e
-  url: https://files.pythonhosted.org/packages/43/5f/e53a850fd32dddefc998b6bfcbda843d4ff5b0dcac02a92e414ba6c97d46/pydantic-1.10.7.tar.gz
+  sha256: 4fd5c182a2488dc63e6d32737ff19937888001e2a6d86e94b3f233104a5d1fa9
```
Comment (hoodmane) [inline packages/pydantic/meta.yaml:7]: This might not work, huh? We are missing `pydantic-core` which I think is now a rust package?

Comment (ryanking13) [reply packages/pydantic/meta.yaml:7]: Yes, need to build `pydantic-core` to make pydantic 2.x work.

File: packages/shapely/meta.yaml (hunk 1 above)
Comment (ryanking13) [reply packages/shapely/meta.yaml:10]: That patch needs to be dropped. Let me see if I can update shapely.

**Comment by ianthomas23** [comment]
If you are updating matplotlib to 3.8.2 then it should also have contourpy as a new runtime dependency (see https://github.com/matplotlib/matplotlib/blob/4591c5b8c5a7b8b721f8173ba5d7ed7751fe6922/setup.py#L332). If you don't want to do it in this PR I will submit a separate one for it afterwards.

**Comment by hoodmane** [comment]
Thanks for pointing that out @ianthomas23, I added it.

**Comment by ryanking13** [comment]
Thanks for updating the packages!

**Pull request merged**